lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -1556,21 +1557,30 @@ chooseMfmaLikeStoreLayout(RankedTensorType valType) {
   // We make each thread handle 8 consecutive elements to enable 128-bit
   // global stores for [b]f16 types and keep the thread pattern in each lane
   // similar to the canonical mfmaLayout.
-  LinearLayout mfma8Layout = LinearLayout::empty();
-  mfma8Layout =
-      LinearLayout({{kRegister, {{1, 0}, {2, 0}, {4, 0}}},
-                    {kLane, {{0, 1}, {0, 2}, {0, 4}, {0, 8}, {0, 16}, {8, 0}}},
-                    {kWarp, {}},
-                    {kBlock, {}}},
-                   {standardOutDims[order[0]], standardOutDims[order[1]]});
+  LinearLayout storeLayout = LinearLayout::empty();
+  if (isMfma32) {
+    storeLayout = LinearLayout(
+        {{kRegister, {{1, 0}, {2, 0}, {4, 0}}},
+         {kLane, {{0, 1}, {0, 2}, {0, 4}, {0, 8}, {0, 16}, {8, 0}}},
+         {kWarp, {}},
+         {kBlock, {}}},
+        {standardOutDims[order[0]], standardOutDims[order[1]]});
+  } else {
+    storeLayout = LinearLayout(
+        {{kRegister, {{1, 0}, {2, 0}}},
+         {kLane, {{0, 1}, {0, 2}, {0, 4}, {0, 8}, {0, 16}, {4, 0}}},
```

Comment:
I'm not sure this is correct still--I'd expect we have 8 consecutive elements in registers so like L1563. For lane arrangement, we should only have 16 threads consecutive, see L465 but we have 32 threads here?